components/server/src/prebuilds/github-enterprise-app.ts

```diff
@@ -168,15 +168,18 @@ export class GitHubEnterpriseApp {
             });
 
             const config = await this.prebuildManager.fetchConfig({ span }, user, context, project?.teamId);
-            if (
-                !this.prebuildManager.shouldPrebuild({ config, project }) ||
-                !this.appRules.shouldRunPrebuild(config, context.ref === context.repository.defaultBranch, false, false)
-            ) {
+            const prebuildPrecondition = this.prebuildManager.checkPrebuildPrecondition({ config, project, context });
+
+            const shouldRun = Project.hasPrebuildSettings(project)
```

Comment:
why is this check not part of `checkPrebuildPrecondition`?

Reply:
I see you use this to check if we need to call appRules. We could move them to `checkPrebuildCondition` as well as they seem to also only use the passed-in state (config).
OTOH I understood we would simple deprecate and remove the appRules config. Why don'T we do that already?

Reply:
As we are to remove appRules, which are also marked as deprecated here, I wanted to make it easier to remove those parts. This means, the new check has to be complete if `appRules` are removed. But this also means, for pre-existing projects appRules should continue working. Once the data migration is done, this code paths are no longer used and we can remove it easily.